[PXCT-1735] - Edge AI Course (#2837)
arduino/docs-content: content/learn/09.edge-ai/03.eac3-edge-ai-workflow/content.md

|        Image classification        |              100              |            500 - 1,000            |

**These figures are guidelines**. In general, the more similar the classes are to each other (for example, distinguishing between two similar gestures), the more data the model needs to learn the differences between them.

#### Balance Between Classes

An imbalanced dataset, where some classes have many more samples than others, can bias the model toward the classes with more examples. For example, if there are 500 samples of “walking” but only 50 of “running”, the model will tend to classify everything as “walking” because that is the correct answer most of the time [7].

It is recommended to maintain a similar number of samples in each class. When an imbalance is unavoidable (for example, in anomaly detection, where abnormal events are rare), we can use compensation techniques discussed in the preprocessing section.

#### Dataset Split

The collected data must be divided into three sets with distinct purposes:

- **Training set (60-80%)**: The dataset from which the model learns. The model adjusts its internal parameters to recognize the patterns present in this data.

> - **Validation set (10-20%)**: Used during training to verify that the model is not simply memorizing the data. It allows detecting overfitting problems.

Reviewer: ```suggestion
- **Validation set (10-20%)**: Used during training to verify that the model is not simply memorizing the data. 
```

#### Data Augmentation for Sensor Signals

Audio signals can undergo specific transformations, such as adding background noise recorded in the deployment environment, slightly adjusting the recording speed, and mixing the signal with other ambient sounds.

For voice command recognition, adding background noise is particularly important because it simulates the real conditions in which the model will operate.

#### Data Augmentation for Audio

Audio signals can undergo specific transformations, such as adding background noise recorded in the deployment environment, slightly adjusting the recording speed, and mixing the signal with other ambient sounds.

For voice command recognition, adding background noise is particularly important because it simulates the real conditions in which the model will operate.

#### Data Augmentation for Images

Images offer a wide range of possible transformations. The most commonly used in Edge AI include slight rotation (typically ±15°), brightness and contrast adjustments, random crops, and horizontal flip (when it makes sense for the application).

> The transformations must be realistic. For example, for a product classifier on a shelf, rotating the image 180° would not make sense because products are always oriented upward.

Reviewer: ```suggestion
The transformations must be realistic. For instance, it would not make sense to rotate a product classifier on a shelf 180° because products are always oriented upward.
```
